src/components/Table/Table.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import {
+  getCoreRowModel,
+  getSortedRowModel,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table';
+import { Flex, ScrollView, FlatList, Box } from 'native-base';
+
+import { createTableColumns } from './createTableColumns';
+import { TableHeader } from './TableHeader';
+import { TableRow } from './TableRow';
+
+type Props = {
+  data: object[];
+  onRowClick: (row: Row<any>) => void;
+  minColumnWidth?: number;
```

Comment:
I feel that the minimum column width should be set on a per column basis. Or you could have a global min width that can be overriden. eg.

```
type Props = {
  minWidth: number;
  columnConfig: {
    minWidth: number;
  }
}

...

<TableHeader
  minWidth={columnConfig?.[columnId]?.minWidth ?? minWidth}
/>
```

Maybe not needed for now, but a thought for the future